### PR TITLE
[EDIT] Go: Composition - Inheritance examples

### LINKED ISSUE #3400: [Edit] Go [Edit] Go: Composition
### Type of Edit (select all that apply)

Add new content (definitions, codeblocks, etc.)

### Description (optional)

This edit is for the Go [Composition](https://www.codecademy.com/resources/docs/go/composition) located under the file path `docs/content/go/concepts/composition/composition.md` 

Currently, the entry does not show struct embedding and how the fields and methods of the embedded type are inherited by the created type. Be sure to include:

- A "Syntax" section with a ` ```pseudo` code snippet
- An "Example" section with a ` ```go` code snippet

Please refer to the [entry templates](https://github.com/Codecademy/docs/blob/main/documentation/), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Jamesllllllllll**
### Description

Added examples of inheriting methods and fields with `struct` embedding. Added `pseudo` code block, a brief `go` code snippet showing how to add it to Pizza Restaurant example, a final `go` code snippet with the full example and the output.

Closes #3400 

### Type of Change

- Editing an existing entry (adding examples of inheritance via composition)

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/composition/composition.md [hunk 1]
```diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
 Pizzeria del Corso has a rating of 4 and serves the following pizzas: [{Margherita medium [tomatoes mozzarella basil] true}]
 ```
 
+## Inheritance through Composition
+
+A `struct` can 'inherit' the fields and methods of an embedded `struct` through composition rather than traditional [inheritance](https://www.codecademy.com/resources/docs/general/programming-paradigms/inheritance).
```
Comment (caupolicandiaz) [inline content/go/concepts/composition/composition.md:138]:
```suggestion
A struct can 'inherit' the fields and methods of an embedded struct through composition rather than traditional [inheritance](https://www.codecademy.com/resources/docs/general/programming-paradigms/inheritance).
```

Comment (caupolicandiaz) [reply content/go/concepts/composition/composition.md:138]: The struct referenced here doesn't need to be inside backticks. This formatting is used to highlight something referenced in the code examples, such as a variable. Or to offset specific syntax, such as declaring a struct requires the `struct` keyword. This formatting is incorrectly used several times in the entry, please revert the other examples as well.

File: content/go/concepts/composition/composition.md [hunk 2]
```diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
...
+type MainStruct struct {
+  Embedded
+}
+
+func PrintEmbedded(m MainStruct) string {
+  return m.printField()
+}
+```
+
+Consider the following addition to the Pizza Restaurant example:
```
Comment (caupolicandiaz) [inline content/go/concepts/composition/composition.md:158]:
```suggestion
Consider the following addition to the pizza example:
```

File: content/go/concepts/composition/composition.md [hunk 3]
```diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
...
+  PizzaMenu []Pizza
+  Address
+}
+
+func restaurantInfo(r Restaurant) string {
+  return r.Name + ", located at " + r.printAddress() + ", " + r.Country + " has a rating of " + fmt.Sprint(r.Rating) + " and serves the following pizzas: " + fmt.Sprint(r.PizzaMenu)
+}
+```
+
+An `Address` `struct` is created, and a `printAddress()` method takes an `Address` as its receiver. The anonymous field `Address` is added to `Restaurant` which allows access to the `Address` fields and methods.
```
Comment (caupolicandiaz) [inline content/go/concepts/composition/composition.md:183]:
```suggestion
An `Address` struct is created, and a `printAddress()` method takes an `Address` as its receiver. The anonymous field `Address` is added to `Restaurant` which allows access to the `Address` fields and methods.
```

File: content/go/concepts/composition/composition.md [hunk 4]
```diff
@@ -133,6 +133,129 @@ Margherita pizza is a medium pizza with toppings of: [tomatoes mozzarella basil]
...
+
+The `restaurantInfo()` function can now access the `printAddress()` method with `r.printAddress()` and the `Country` field with `r.Country`.
+
+The following example adds the `Address` `struct` and method:
+
+```go
```
Comment (caupolicandiaz) [inline content/go/concepts/composition/composition.md:189]: This code is getting is bloated. I think it's preferable to just show the main function and perhaps have a comment at the top that this builds on the previous code

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@Jamesllllllllll I've added my comments. Please update the entry, and tag me when you've finished making the changes. Thanks

File: content/go/concepts/composition/composition.md (hunk 4 above)
Comment (Jamesllllllllll) [reply content/go/concepts/composition/composition.md:189]: How about this wording for the comment?

**Comment by Jamesllllllllll** [comment]
> @Jamesllllllllll I've added my comments. Please update the entry, and tag me when you've finished making the changes. Thanks

Thank you for your review @caupolicandiaz ! 

Check the comment I added - I wasn't sure if I should capitalize Struct and Method, or if that is the best wording for the comment.

**Review by caupolicandiaz — APPROVED** [review]
@Jamesllllllllll Just made a couple of additional direct edits. This is ready for a second review

**Review by KTom101 — APPROVED** [review]
@Jamesllllllllll This looks good.
